### PR TITLE
Commitment -> Post Data

### LINKED ISSUE #39: Rename "Commitment"
Tal takes offense at our choice of the word "Commitment" (since it has a specific, cryptographic meaning elsewhere in our protocol). Let's rename the types/messages to PostData, AllocatedData, or something else - maybe DiskSpaceAllocatedToSpacemesh? :)

### PR DESCRIPTION
**Pull request by avive**
Improve ComputeEngine selection to match current caps of gpu-post.

Closes #39

### CODE REVIEW

File: proto/spacemesh/smesher.proto [hunk 1]
```diff
@@ -36,26 +36,25 @@ service SmesherService {
     // Returns true if request succeeds, false if it fails.
     rpc SetMinGas(SimpleInt) returns (TypedResult);
 
-    /////// PoSpace commitments management
+    /////// Proofs of space data files management
 
-    // Returns post commitment data from the node
-    rpc CommitmentData(google.protobuf.Empty) returns (CommitmentData);
+    // Returns post data status from the node
+    rpc PostData(google.protobuf.Empty) returns (PostStatus);
```
Comment (lrettig) [inline proto/spacemesh/smesher.proto:42]:
```suggestion
    rpc PostStatus(google.protobuf.Empty) returns (PostStatus);
```

File: proto/spacemesh/smesher_types.proto [hunk 2]
```diff
@@ -1,55 +1,52 @@
...
-message CommitmentProcessors {
-    repeated CommitmentProcessor processors = 1;
+// Basic post data. Used for users to provide requested post data and by
+// other messages which contain post data.
+message PostData {
+    string path = 1; // User provided path to create the post data files at
+    uint64 data_size = 2; // Requested post data size
+    bool   append = 3; // Append to existing files if they exist. Otherwise overwrite.
+    bool   throttle = 4; // Throttle down setup phase computations while user is interactive on ///system
+    uint32 compute_engine_flags = 5; // One ore more available compute engines to be used for gpu post
```
Comment (lrettig) [inline proto/spacemesh/smesher_types.proto]:
```suggestion
    ComputeEngineFlags flags = 5; // One ore more available compute engines to be used for gpu post
```

Comment (avive) [reply proto/spacemesh/smesher_types.proto]: Unfortunately, this won't work as the ComputeEngineFlags is just an enum and the uint32 is needed to represent a bit field of one or more of these flags set. I wish had support for this but it doesn't. Used best practices after reading on bit fields in protobufs...

Comment (avive) [reply proto/spacemesh/smesher_types.proto]: updated

**Pull request merged**